librz/analysis/p/analysis_bf.c

```diff
@@ -44,7 +45,9 @@ static void bf_syscall_write(RzILVM *vm, RzILOpEffect *op) {
 	}
 	RzBitVector *bv = rz_il_vm_mem_load(vm, 0, ptr_val->data.bv);
 	ut32 c = rz_bv_to_ut32(bv);
-	putchar(c);
+	if (c) {
+		rz_cons_printf("%c", c);
+	}
```

Comment:
I think analysis plugins (nor even analysis module itself) should have a dependency on rz_cons.

Reply:
Should not?

Reply:
Should not, ofc :D

Reply:
Ok I'll see how I can pass a callback to the plugin

Reply:
Why the analysis plugin has to write stuff on the console?

Reply:
It's part of brainfuck emulation, which has a print instruction.

Reply:
:( I see. What would be a solution then? Just depend on rz_cons?

Reply:
52b042706cead4b5029f4c3f31c1e7bbaf17fa49 is a solution

Reply:
Kinda ok for now, though ideally I'd love to have RzAnalysis, RzAsm, RzFlag, etc. without that "cb_printf" callback...  Not sure if it will be fully possible.